app/models/mongoidable/abilities.rb

```diff
@@ -17,7 +17,7 @@ def initialize(ability_source, parent_model)
 
     def cannot?(*args)
       if can_cache?
-        Rails.cache.fetch(ability_cache_key(args), cache_options) { super }
+        !Rails.cache.fetch(ability_cache_key(args), cache_options) { !super }
```

Comment:
We're only caching the `can?` value.  `cannot?` has to read the opposite of the cached value.